systest/online-restore/online_restore_test.go

```diff
@@ -222,6 +239,107 @@ func TestBasicRestore(t *testing.T) {
 	runMutations(t, dg)
 }
 
+// The 6 backups that are being restored in this test were taken in the following manner
+// Add _:a <name> "alice" .
+// Take backup b1
+// Add _:b <name> "bob" .
+// Add _:b <age> "12" .
+// Take backup b2
+// Drop attribute "name"
+// Take backup b3
+// Add _:a <name> "alice" .
+// Take backup b4
+// drop data
+// Take backup b5
+// drop all
+// Take backup b6
+func TestIncrementalRestore(t *testing.T) {
+	conn, err := grpc.Dial(testutil.SockAddr,
+		grpc.WithTransportCredentials(credentials.NewTLS(testutil.GetAlphaClientConfig(t))))
+	require.NoError(t, err)
+	dg := dgo.NewDgraphClient(api.NewDgraphClient(conn))
+
+	ctx := context.Background()
+	require.NoError(t, dg.Alter(ctx, &api.Operation{DropAll: true}))
+
+	req := &restoreReq{
+		location:  backup2011Location,
+		backupNum: 1,
+	}
+	sendRestoreRequest(t, req)
+	testutil.WaitForRestore(t, dg, testutil.SockAddrHttp)
+
+	query := `{ q(func: has(name)) { name age } }`
+	runQuery := func(query, expectedResp string) {
+		res, err := dg.NewTxn().Query(context.Background(), query)
+		require.NoError(t, err)
+		require.JSONEq(t, expectedResp, string(res.Json))
+	}
+	runQuery(query, `{"q":[{"name":"alice"}]}`)
+
+	req = &restoreReq{
+		location:        backup2011Location,
+		backupNum:       2,
```

Comment:
Why is `backupNum` same as `incrementalFrom`?

Reply:
backupNum is the backupNum of the backup that we are restoring up to in the series of backups. We are just restoring one incremental backup at a time.

Reply:
Is it correct to understand that the backup happens in the following range:

`incrementalFrom<= xyz <= backupNum`. 

I am curious if we can add a test where these two numbers (`incrementalFrom` and `backupNum` are not identical.)

Reply:
I'm working on such a test as part of #8754. The diff of the new PR doesn't look good yet because it depends on this and one more PR.